staging/src/k8s.io/csi-api/pkg/crd/crd.go

```diff
@@ -31,12 +32,27 @@ import (
 func CSIDriverCRD() *apiextensionsv1beta1.CustomResourceDefinition {
 	return &apiextensionsv1beta1.CustomResourceDefinition{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: csiapiv1alpha1.CsiDriverResourcePlural + "." + csiapiv1alpha1.GroupName,
+			Name: csiapiv1beta1.CsiDriverResourcePlural + "." + csiapiv1beta1.GroupName,
+		},
+		Status: apiextensionsv1beta1.CustomResourceDefinitionStatus{
+			StoredVersions: []string{"v1beta1"},
 		},
 		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
-			Group:   csiapiv1alpha1.GroupName,
-			Version: csiapiv1alpha1.SchemeGroupVersion.Version,
-			Scope:   apiextensionsv1beta1.ClusterScoped,
+			Group:   csiapiv1beta1.GroupName,
+			Version: csiapiv1beta1.SchemeGroupVersion.Version,
+			Versions: []apiextensionsv1beta1.CustomResourceDefinitionVersion{
+				{
+					Name:    "v1beta1",
+					Served:  true,
+					Storage: true,
+				},
+				{
+					Name:    "v1alpha1",
+					Served:  true,
```

Comment:
Would this break the upgrade path for those who are using the current alpha version?

Reply:
what upgrade path are you referring to? leaving an unserved/unstored alpha version listed would allow someone currently using the alpha version to update to this version without first deleting all their alpha data. the objects would be available via the beta endpoints, and on next write would be stored as beta objects.